cmd/talosctl/cmd/talos/upgrade-k8s.go

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package talos
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+
+	"github.com/talos-systems/talos/pkg/cli"
+	"github.com/talos-systems/talos/pkg/cluster"
+	k8s "github.com/talos-systems/talos/pkg/cluster/kubernetes"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+)
+
+// upgradeK8sCmd represents the upgrade-k8s command.
+var upgradeK8sCmd = &cobra.Command{
+	Use:   "upgrade-k8s",
+	Short: "Upgrade Kubernetes control plane in the Talos cluster.",
+	Long:  `Command runs upgrade of Kubernetes control plane components between specified versions. Pod-checkpointer is handled in a special way to speed up kube-apisever upgrades.`,
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return WithClient(upgradeKubernetes)
+	},
+}
+
+var upgradeK8sCmdFlags struct {
+	fromVersion string
+	toVersion   string
+}
+
+func init() {
+	upgradeK8sCmd.Flags().StringVar(&upgradeK8sCmdFlags.fromVersion, "from-version", "", "the Kubernetes control plane version to upgrade from")
+	upgradeK8sCmd.Flags().StringVar(&upgradeK8sCmdFlags.toVersion, "to-version", constants.DefaultKubernetesVersion, "the Kubernetes control plane version to upgrade to")
```

Comment:
super nit: The `-version` is a little bit redundant since the description explains the flag. We could maybe simplify this to

```
--to
--from
```

Not sold completely on my own idea, but just an idea.

Reply:
yep, I had some idea. I thought it might be easier to add another `--from-xxx` if we ever need, but not sure if there might be any other flag.

Reply:
updated to `--from`